packages/inter-protocol/src/auction/offerBook.js

```diff
@@ -126,90 +216,145 @@ export const prepareScaledBidBook = baggage =>
+        bidDataKitPromises.init(key, bidDataKitP.promise);
+        E.when(makeBidNode(bidsNode, seqNum), childBidNode => {
+          const recorderKit = makeRecorderKit(childBidNode);
+          bidDataKits.init(key, recorderKit);
+          bidDataKitP.resolve(recorderKit);
+          bidDataKitPromises.delete(key);
+          return recorderKit;
+        });
+
         /** @type {BidderRecord} */
-        const bidderRecord = {
+        const bidderRecord = harden({
           bidScaling: undefined,
           price,
-          received: empty,
+          received: AmountMath.makeEmpty(collateralBrand),
           seat,
           seqNum,
-          wanted,
+          originalWant: wanted,
+          remainingWant: wanted,
           exitAfterBuy,
-        };
-        records.init(key, harden(bidderRecord));
+          timestamp,
+        });
+        records.init(key, bidderRecord);
+        this.self.publishOffer(bidderRecord);
         return key;
       },
       offersAbove(price) {
         const { records } = this.state;
         return [...records.entries(M.gte(toPartialOfferKey(price)))];
       },
+      publishOffer(record) {
+        const key = toPriceOfferKey(record.price, record.seqNum);
+
+        // users can exit seats
+        if (record.seat.hasExited()) {
+          this.self.delete(key);
+          return;
+        }
+
+        return E(getBidDataRecorder(key)).write(
+          harden({
+            price: record.price,
+            originalWant: record.originalWant,
+            remainingWant: record.remainingWant,
+            exitAfterBuy: record.exitAfterBuy,
+            timestamp: record.timestamp,
+            balance: record.seat.getCurrentAllocation().Bid,
+            sequence: record.seqNum,
+          }),
+        );
+      },
       hasOrders() {
         const { records } = this.state;
         return records.getSize() > 0;
       },
       delete(key) {
         const { records } = this.state;
+        void deleteNodeIfPresent(key);
+        bidDataKits.delete(key);
         records.delete(key);
       },
       updateReceived(key, sold) {
         const { records } = this.state;
         const oldRec = records.get(key);
-        records.set(
-          key,
-          harden({
-            ...oldRec,
-            received: AmountMath.add(oldRec.received, sold),
-          }),
-        );
+        const newRecord = harden({
+          ...oldRec,
+          received: AmountMath.add(oldRec.received, sold),
+          remainingWant: AmountMath.subtract(oldRec.remainingWant, sold),
+        });
+        records.set(key, newRecord);
+        this.self.publishOffer(newRecord);
       },
       exitAllSeats() {
         const { records } = this.state;
         for (const [key, { seat }] of records.entries()) {
           if (!seat.hasExited()) {
             seat.exit();
+            if (bidDataKits.has(key)) {
+              bidDataKits.delete(key);
+            }
+            void deleteNodeIfPresent(key);
```

Comment:
it seems a bit fragile to have to do both of these delete operations here. It would be nice to wrap the `bidDataKits` collection in something that automatically updates the storageNodes as appropriate. But maybe the values in the collection aren't enough info to update the storage nodes. hm.

Reply:
I would have liked to just call `delete(key)`, but that is currently only called when it's known that the record and node were actually created.   Aah! I suppose I can put the conditional in that fn at no risk and little cost.